types/mongoose/index.d.ts

```diff
@@ -582,6 +528,9 @@ declare module "mongoose" {
     /** Adds key path / schema type pairs to this schema. */
     add(obj: SchemaDefinition, prefix?: string): void;
 
+    /** Return a deep copy of this schema */
+    clone(): Schema;
```

Comment:
Return `this` to be consistent with other methods?

Reply:
It's semantics at this point. `this` refers to the same object, whereas here we're returning a whole new one.

Reply:
Oh okay, my thinking was that returning `Schema` returns the base `Schema` class. If we clone, we could be cloning on an interface that extends the base `Schema` class, so returning `this` would be more accurate.

I think returning `this` means returning the same interface that `clone()` was called on rather than returning the same `object` like in normal code.